src/App.jsx

```diff
@@ -1,107 +1,32 @@
 import './App.scss';
 
-// import postsFromServer from './api/posts.json';
-// import commentsFromServer from './api/comments.json';
-// import usersFromServer from './api/users.json';
+import { PostList } from './components/PostList';
 
-export const App = () => (
-  <section className="App">
-    <h1 className="App__title">Static list of posts</h1>
-
-    <div className="PostList">
-      <div className="PostInfo">
-        <div className="PostInfo__header">
-          <h3 className="PostInfo__title">qui est esse</h3>
-
-          <p>
-            {' Posted by  '}
-
-            <a className="UserInfo" href="mailto:Sincere@april.biz">
-              Leanne Graham
-            </a>
-          </p>
-        </div>
-
-        <p className="PostInfo__body">
-          est rerum tempore vitae sequi sint nihil reprehenderit dolor beatae
-          ea dolores neque fugiat blanditiis voluptate porro vel nihil
-          molestiae ut reiciendis qui aperiam non debitis possimus qui neque
-          nisi nulla
-        </p>
-
-        <hr />
-
-        <b data-cy="NoCommentsMessage">No comments yet</b>
-      </div>
-
-      <div className="PostInfo">
-        <div className="PostInfo__header">
-          <h3 className="PostInfo__title">
-            doloremque illum aliquid sunt
-          </h3>
+import postsFromServer from './api/posts.json';
+import commentsFromServer from './api/comments.json';
+import usersFromServer from './api/users.json';
 
-          <p>
-            {' Posted by  '}
+const posts = postsFromServer
+  .map(post => ({
+    ...post,
+    comments: getCommentsById(post.id),
+    user: getUserById(post.userId),
+  }));
 
-            <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-              Patricia Lebsack
-            </a>
-          </p>
-        </div>
+function getCommentsById(id) {
+  return commentsFromServer.filter(comment => comment.postId === id) || null;
```

Comment:
if you want to process null from backend it is better to return an empty array, because in markup you check if `array.length !== 0`, then if you filter array and there is no match, filter returns you an empty array. No need to use null here as default value

```suggestion
  if(!commentsFromServer) {
  return [];
  }
  return commentsFromServer.filter(comment => comment.postId === id);
```